api/src/main/java/org/apache/cloudstack/api/command/admin/network/CreateNetworkOfferingCmd.java

```diff
@@ -338,10 +340,15 @@ public Map<String, List<String>> getServiceProviders() {
     private void getServiceProviderMapForNsx(Map<String, List<String>> serviceProviderMap) {
         String routerProvider = Boolean.TRUE.equals(getForVpc()) ? VirtualRouterProvider.Type.VPCVirtualRouter.name() :
                 VirtualRouterProvider.Type.VirtualRouter.name();
-        List<String> unsupportedServices = List.of("Vpn", "SecurityGroup", "Connectivity",
-                "Gateway", "Firewall", "BaremetalPxeService");
+        List<String> unsupportedServices = new ArrayList<>(List.of("Vpn", "SecurityGroup", "Connectivity",
```

Comment:
Why new ArrayList here?

Reply:
To create a service map for NSX - coz we predetermine the services NSX offering will offer.. and we want to filter out other services not supported